### ISSUE TITLE
U+2571 U+2572 U+2573 box drawings slope problem

### ISSUE THREAD

**Issue by maverickwoo** [cormullion/juliamono#145]
I am using font version 0.045, and I confirm that this is reproducible on macOS 12.5 with VS Code 1.69.2 using a minimal `settings.json` pasted below. All three versions are the latest as of this writing.

```json
{
  "editor.fontFamily": "JuliaMono",
  "editor.fontSize": 14
}
```

Using this setup, I observe that the slopes of the strokes in the following three code points for box drawings are approximately ±45 degrees.

U+2571 `╱` Box Drawings Light Diagonal Upper Right To Lower Left
U+2572 `╲` Box Drawings Light Diagonal Upper Left To Lower Right
U+2573 `╳` Box Drawings Light Diagonal Cross

Unfortunately, this choice considerably reduces the utility of these code points for box drawings because the diagonal box lines drawn with them do not visually align. To illustrate this, consider the demo file at https://www.cl.cam.ac.uk/~mgk25/ucs/examples/UTF-8-demo.txt. Here is the source code of one of the boxes towards the end of the file:

```
╔══╦══╗ 
║┌─╨─┐║ 
║│╲ ╱│║ 
╠╡ ╳ ╞╣ 
║│╱ ╲│║ 
║└─╥─┘║ 
╚══╩══╝ 
```

And here is a screenshot of this box drawn by VS Code in the above setup:

<img width="72" alt="Screen Shot 2022-07-24 at 18 37 45" src="https://user-images.githubusercontent.com/172693/180668647-25479b22-d492-473e-b730-49f092049f7d.png">

Perhaps it would be an improvement to make the slopes in these three code points steeper (and with suitable adjustments to the stroke lengths) such that the diagonal box lines will visually align with each other?

Thank you very much for your consideration. I really appreciate the effort you have put into this fantastic font.

**Comment by cormullion** [comment]
Hi! This is certainly possible to do, and I've tried making the changes.

However, it will be difficult to guarantee that it looks good in all situations; it depends on the line height. If you can adjust the line height of your editor, you can adjust it to look reasonably OK.

In VSCode it looks better in the terminal (line height 1) than in the editor (line height is obviously > 1).

<img width="376" alt="Screenshot 2022-07-25 at 17 49 23" src="https://user-images.githubusercontent.com/52289/180831912-462f3243-1de7-4ed3-90a6-9c2ec5a6ca98.png">

In a MacOS Terminal, you can adjust the line height precisely so you can make the diagonals line up.

<img width="640" alt="Screenshot 2022-07-25 at 17 58 36" src="https://user-images.githubusercontent.com/52289/180833554-bb3b4205-260a-4381-9f0d-445b65bb57e6.png">

Unlike for vertical and horizontal lines, the diagonals can't overshoot/overlap for every setting of line spacing.

Here in Github, using the github font:
```
╔══╦══╗ 
║┌─╨─┐║ 
║│╲ ╱│║ 
╠╡ ╳ ╞╣ 
║│╱ ╲│║ 
║└─╥─┘║ 
╚══╩══╝ 

╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱ 
╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
 ╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
 ╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
 ╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╳╲╱ 
```

**Comment by maverickwoo** [comment]
Thank you very much for the explanations. I acknowledge some line spacing adjustment may be needed to make the new slope-adjusted and lengthened diagonal strokes line up.

If I may make a suggestion, perhaps the documentation can pick a program and a line spacing and state that particular setup is the "benchmark", i.e., the strokes in the released font files are intended to line up in that one tested setting and line spacing adjustment may be needed in other programs to achieve the same result. This way, all future users can know whether they are seeing a bug or not.

Thank you very much for the fixes. I look forward to seeing the next release.

**Comment by DanielMGessel** [comment]
Apologies for jumping in here. I was just playing around with an SVG file I hacked up and noticed the block characters seem really tall. I was about to submit a new issue when I realized there is no fixed standard for line spacing! I was assuming that a monospace font should be designed to be spaced "font height" apart (1em); it seems obvious that monospace should apply to the vertical as well as horizontal (to an old hacker like me, anyway). A little more searching stopped me in my tracks when I discovered recommendations from 1.2em to 1.5em!

Using 1em spacing, the ratio of vertical to horizontal comes out to be about 15 to 9? So, for the block characters, what's the intended line spacing for JuliaMono? Closer to 1.5em?

Well, here's that SVG if you want to play with it - apologies that it's just a junk file... I wonder if there are reference images for a set of SVGs to test rendering conformance? We always used reference images for graphics driver work.

```
<svg width="2in" height="2in"
     preserveAspectRatio = "none"
     viewBox="0 0 90 90"
     xmlns="http://www.w3.org/2000/svg"
     style="background: #000; fill:#000;">
<g transform="translate(0 90) scale(1 -1.0)">
<style>
@font-face {
  font-family: JuliaMono;
  src: url("https://cdn.jsdelivr.net/gh/cormullion/juliamono/webfonts/JuliaMono-Bold.woff2");
  font-weight: normal;
}
.f
{font:bold 10px "JuliaMono";}
.j
{font:bold 10px "JuliaMono";}
</style>
<rect x="0" y="0" width="90" height="90" style="fill:#024;stroke:#036;stroke-width:3px" />
<text transform="translate(0 90) scale(1 -1)"
  class="f" text-anchor="left" stroke-width="0" style="fill: #FFF; line-height: normal">
<tspan class="f" x="0" dy="1lh">0123456789ABCDEF</tspan>
<tspan class="j" x="0" dy="1lh">0123456789ABCDEF</tspan>
<tspan class="f" x="0" dy="1lh">ABCabc</tspan><tspan class="j">ABCabcmmmmlmmmmmmmmmmm</tspan>
<tspan class="f" x="0" dy="1em">𝙰𝙱𝙲𝙳𝙴𝙵𝙶</tspan><tspan class="j">𝙰𝙱𝙲𝙳𝙴𝙵𝙶𝙷𝙸𝙹𝙺𝙻𝙼𝙽3mm ┃mmmmmmmmmmmm</tspan>
<tspan class="f" x="0" dy="1em">𝒪𝒫𝒬ℛ𝒮𝒯</tspan><tspan class="j">𝒪𝒫𝒬ℛ𝒮𝒯𝒰𝒱𝒲4mm┃ mmmmmmmmm</tspan>
<tspan class="j" x="0" dy="1em">5mm mm┃mmmmmmm</tspan>
<tspan class="f" x="0" dy="1em">6mmmmmmmmmmmm</tspan>
<tspan class="f" x="0" dy="1em">7mmmmmmmmmmmm</tspan>
<tspan class="f" x="0" dy="1em">8wait MMM what </tspan>
<tspan class="f" x="0" dy="1em">9wait what </tspan>
<tspan class="f" x="0" dy="1em"> wait what </tspan>
<tspan class="f" x="0" dy="1em"> wait what </tspan>
<tspan class="f" x="0" dy="1em"> wait what </tspan>
</text>
</g>
</svg>
```

**Comment by cormullion** [comment]
Yes - I think of an OTF font file as being like the score for a piece of music - it gets interpreted by different performers with different results, and who's to say whether there's a definitive performance.. 😀

**Comment by anderslundstedt** [comment]
In iTerm I can get box characters to line up correctly at the maximum line spacing of 200:

<img width="1440" alt="iterm" src="https://user-images.githubusercontent.com/4514101/192833302-d436b2fe-b690-4986-88f4-3ae30021e377.png">

In verbatim environments in XeLaTeX I can get box characters to line up correctly up to a line spacing of 1.4:

<img width="590" alt="xelatex" src="https://user-images.githubusercontent.com/4514101/192833814-452b11ca-4729-49a6-b570-7a73a1407e70.png">

At 1.5 they almost line up, but there seems to be an issue with at least the glyph for ‘┐’ but probably also with the glyph for ‘┘’:

<img width="1285" alt="xelatex_zoom" src="https://user-images.githubusercontent.com/4514101/192834230-095d91c5-8895-4c63-9840-0a6d50478418.png">

**Comment by DanielMGessel** [comment]
Maybe I’ll look at the files a bit more closely…

**Comment by anderslundstedt** [comment]
> In iTerm I can get box characters to line up correctly at the maximum line spacing of 200:
> […]
> 
> In verbatim environments in XeLaTeX I can get box characters to line up correctly up to a line spacing of 1.4:
> [...]

And in TextEdit version 1.17 (380.2) on MacOS Monterey 12 I cannot even make the box characters connect at a line spacing of 1.1 (but 1.0 stil works):

<img width="637" alt="textedit_1 0" src="https://user-images.githubusercontent.com/4514101/192891372-53449a00-8a72-46f6-a510-acf4dffc1dd0.png">

<img width="636" alt="textedit_1 1" src="https://user-images.githubusercontent.com/4514101/192891382-66450d5a-e1fb-4ea0-9cef-f25bebe0aeac.png">

<img width="635" alt="textedit_1 3" src="https://user-images.githubusercontent.com/4514101/192891396-a0debb1b-6020-4a3d-9822-6e58976c47b1.png">

**Comment by DanielMGessel** [comment]
It'd take some deep digging to find out, but I wonder if some render paths are clipping the glyphs.
